packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -373,6 +371,22 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         ).to.equal(200 - 2);
       });
     });
+
+    it('Grid wrapper should have height different from 0 when autoHeight', () => {
+      const { getByRole, getAllByRole } = render(
+        <div style={{ width: 300 }}>
+          <DataGrid {...baselineProps} autoHeight />
+        </div>,
+      );
+
+      const colHeaderRowHeight = getAllByRole('row')[0].clientHeight;
+      const totalRowsHeight = getAllByRole('row')[1].clientHeight * 3;
+      const footerHeight = document.querySelector('.MuiDataGrid-footer')?.clientHeight;
+
+      expect(getByRole('grid').clientHeight).to.equal(
+        colHeaderRowHeight + totalRowsHeight + footerHeight + 1, // The 1 is to compensate, needs to be fixed
+      );
```

Comment:
Did you consider this approach?
```suggestion
      render(
        <div style={{ width: 300 }}>
          <DataGrid {...baselineProps} headerHeight={40} rowHeight={30} autoHeight />
        </div>,
      );
      expect(document.querySelector('.MuiDataGrid-main')!.clientHeight).to.equal(
        40 + 30 * baselineProps.rows.length,
      );
```

Reply:
No, I wanted to make it as OTB as possible, I considered hiding the footer. I'm ok with either as long as it tests what is needed but I dislike the `1` mostly because I cant seem to find where it is coming from (the difference of 1px). Also for the `autoHeight` I was considering the height of the entire grid but this is also ok since it checks the dynamic part of it.

Reply:
I wouldn't be surprised if the 1px comes from the border CSS.

I have made this counter proposal because when we use `autoHeight`, the grid doesn't rely on CSS layout to determine it's height. Instead, it computes it. So while the first version of the test is valid, it doesn't test that the final computed height is the one expected. The proposed version do.